java/kotlin-extractor2/src/main/kotlin/entities/Expression.kt

```diff
@@ -2547,6 +2553,61 @@ private fun KotlinFileExtractor.extractVariableExpr(
     }
 }
 
-private fun KotlinFileExtractor.useVariable(v: KtProperty): Label<out DbLocalvar> {
+private fun KotlinFileExtractor.useVariable(v: KaVariableSymbol): Label<out DbLocalvar> {
     return tw.getVariableLabelFor<DbLocalvar>(v)
 }
+
+context(KaSession)
+fun KotlinFileExtractor.extractReferenceExpression(
+    ref: KtReferenceExpression,
+    enclosingCallable: Label<out DbCallable>,
+    stmtExprParent: StmtExprParent
+) {
+    val exprParent = stmtExprParent.expr(ref, enclosingCallable)
+
+    when (val resolvedCall = ref.resolveCallTarget()) {
+        is KaSimpleVariableAccessCall -> {
+            when (val varSymbol = resolvedCall.symbol) {
+                is KaPropertySymbol -> {
+                    val (target, args) = when (val access = resolvedCall.simpleAccess) {
+                        is KaSimpleVariableAccess.Read -> Pair(varSymbol.getter, listOf())
+                        is KaSimpleVariableAccess.Write -> Pair(varSymbol.setter, listOf(access.value))
+                    }
+
+                    val qualifier: KtExpression? = (ref.parent as? KtDotQualifiedExpression)?.receiverExpression
```

Comment:
After https://github.com/github/codeql/pull/17881 is merged, I think `KtDotQualifiedExpression` needs to be changed to `KtQualifiedExpression`.